File: python_Programmers/1단계/이상한문자만들기.py
# 내 풀이
def solution(s):
    answer = ''
    cnt = 0
    for word in s:
        if word == ' ':
            answer += ' '
            cnt = 0
        elif cnt % 2 == 0:
            answer += word.upper()
            cnt += 1
        else:
            answer += word.lower()
            cnt += 1
    return answer

# 모범풀이
def solution2(s):
    liste = []
    for i in s.split():
        result = ''
        for n, v in enumerate(i):
            if n % 2 == 0:
                result += v.upper()
            else:
                result += v.lower()
        liste.append(result)
    return ' '.join(liste)

s = "try hello world"
print(solution(s))
print(solution2(s))

File: Java_Programmers/1단계/src/이상한문자만들기.java
import java.util.Arrays;

public class 이상한문자만들기 {
    public String solution(String s) {
        String[] str = s.split(" ", -1);
        System.out.println(Arrays.toString(str));
        int num = 0;
        for (String st : str) {
            String temp = "";
            for (int i = 0; i < st.length(); i++) {
                // 짝수
                if (i % 2 == 0) {
                    temp += Character.toUpperCase(st.charAt(i));
                }
                // 홀수
                else {
                    temp += Character.toLowerCase(st.charAt(i));
                }
            }
            str[num++] = temp;
        }
        return String.join(" ", str);
    }

    public String solution2(String s) {
        String answer = "";
        int cnt = 0;
        String[] array = s.split("");
        for(String ss : array) {
            cnt = (ss.equals(" ")) ? 0 : cnt+1;
            answer += cnt % 2 == 0 ? ss.toLowerCase() : ss.toUpperCase();
        }
        return answer;
    }

    public static void main(String[] args) {
        이상한문자만들기 sol = new 이상한문자만들기();
        String s = "try hello world  ";
//        System.out.println(sol.solution(s));
        System.out.println(sol.solution2(s));
    }
}
